### PR TITLE
NIFI-8676 Added 'Tracking Entities' listing strategy to 'ListS3' and 'ListGCSBucket'

### PR DESCRIPTION
**Pull request by tpalfy**
https://issues.apache.org/jira/browse/NIFI-8676

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by QCU87Z** [comment]
Is the intent to get this merged in for the 1.14.1 release? This would hopefully solve an issue we are seeing in our production environment.

**Comment by pvillard31** [comment]
@tpalfy - there is a conflict to resolve

**Review by Lehel44 — APPROVED** [review]

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/ListGCSBucket.java [hunk 1]
```diff
@@ -461,6 +592,84 @@ public void finishListing(final int listCount, final long maxTimestamp, final Se
...
+
+                    if (pageNr != -1 && pageNr != currentPageNr && writer.isCheckpoint()) {
+                        commit(session, listCount);
+                        listCount = 0;
+                    }
+
+                    pageNr = currentPageNr;
+                }
+
+                writer.finishListing();
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/ListGCSBucket.java:640]: It seems ListGCSBucket does not store the listed items but lists all the files again and again.
Maybe `alreadyListedEntities.put(updatedEntity.getIdentifier(), listedEntity);` is missing here (see the same method in ListS3).

**Review by turcsanyip — APPROVED** [review]
Tested and it works also on GCS now.

@tpalfy Thanks for adding this new feature!
@Lehel44 Thanks for the review!

+1 LGTM
Merging to main.

**Comment by jiaxin11** [comment]
When there are a large number of objects in the bucket, the Tracking Entities policy is invalid.
Or because setNextMarker is not called ?

**Pull request closed without merging**